src/main/java/mate/academy/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public class UserDaoImpl implements UserDao {
+    private static SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
```

Comment:
Consider moving the `SessionFactory` initialization to a class-level variable to avoid redundant calls to `HibernateUtil.getSessionFactory()` in each method.